OpenSim/Simulation/Model/Frame.h

```diff
@@ -27,39 +27,58 @@
 #include <OpenSim/Simulation/osimSimulationDLL.h>
 #include <OpenSim/Simulation/Model/ModelComponent.h>
 
-
-
 namespace OpenSim {
 
-class Model;
-
 //=============================================================================
 //=============================================================================
 /**
- * A Frame is a modeling abstraction that defines a Righthanded CoordinateSystem
- * to which components can be attached, such as markers. The Frame allows
- * for a convenient way to describe the location and orientation of such
- * components.
+ * A Frame is an OpenSim abstraction for a reference frame. It consists of 
+ * a right-handed set of three orthogonal axes. Frames are intended to provide
+ * convenient reference frames for locating phyical structures (such as joints
+ * and muscle attachments) as well as provide a convenient basis for performing
+ * spatial calculations such as a frame aligned with the normal direction of a
+ * contact surface and/or located at the center-of-pressure.
+ *
+ * The puprose of a Frame is to provide its Transform (translation of the origin
+ * and the orientation of its axes) in the Ground frame as a function of the
+ * Model's (SimTK::MultibodySystem's) state.
+ *
+ * The Frame class also provides convenience methods for re-expressing vectors
+ * from one Frame to another.
+ *
+ * As already noted, Frames are intended for the purpose of and useful for
+ * locating physical structures such as bodies, their joints, and the locations
+ * where constraints can be connected and forces can be applied. It is perhaps
+ * less evident that Frames can be extremely useful for relating a multitude of
```

Comment:
> This is a good idea, especially since there is no doxygen page for PhysicalOffsetFrame.

Why not? That seems like an important enough class that it deserves some documentation!